### PR TITLE
TF-TRT test Binary op conversion in dynamic shape mode 

### PR DESCRIPTION
**Pull request by tfeher**
This PR adds explicit batch and dynamic shape tests to the Binary op converter. Tagging @bixia1 for review.

### CODE REVIEW

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc [hunk 1]
```diff
@@ -4149,8 +4149,13 @@ Status ConvertBinary(OpConverterParams* params) {
                                    " inputs but expected 2, at ",
                                    node_def.name());
   }
+#if IS_TRT_VERSION_GE(6, 0, 0, 0)
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc:4152]: A better way is to use a static array to hold the data types and only this array value is conditional on TRT version.

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc [hunk 2]
```diff
@@ -2577,46 +2492,59 @@ TEST_F(OpConverterTest, ConvertBinary) {
         "both input as constant at: my_add");
   }
 
+  using OpFunc = std::function<NodeDef(DataType)>;
+  std::map<std::string, std::pair<OpFunc, std::vector<float>>> op_map;
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:2496]: Can we use a more specific name, such as op_to_test_info?

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc [hunk 3]
```diff
@@ -2577,46 +2492,59 @@ TEST_F(OpConverterTest, ConvertBinary) {
...
+  ADD_OP("Div", ops::Div, {1.5, 3, 1, 2, 1.5, 3, 1, 2});
+  ADD_OP("RealDiv", ops::RealDiv, {1.5, 3, 1, 2, 1.5, 3, 1, 2});
+  ADD_OP("FloorDiv", ops::FloorDiv, {1, 3, 1, 2, 1, 3, 1, 2});
+  ADD_OP("Minimum", ops::Minimum, {2, 2, 3, 3, 2, 2, 3, 3});
+  ADD_OP("Maximum", ops::Maximum, {3, 6, 3, 6, 3, 6, 3, 6});
+  ADD_OP("Pow", ops::Pow, {9, 36, 27, 216, 9, 36, 27, 216});
+#undef ADD_OP
+  std::vector<string> ops_to_test;
+  // Add all ops supported by ConvertBinary.
+  auto* map = BinaryOperationMap();
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:2514]: Let's rename it to supported_ops.

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc [hunk 4]
```diff
@@ -2577,46 +2492,59 @@ TEST_F(OpConverterTest, ConvertBinary) {
...
+  ADD_OP("Sub", ops::Sub, {1, 4, 0, 3, 1, 4, 0, 3});
+  ADD_OP("Mul", ops::Mul, {6, 12, 9, 18, 6, 12, 9, 18});
+  ADD_OP("Div", ops::Div, {1.5, 3, 1, 2, 1.5, 3, 1, 2});
+  ADD_OP("RealDiv", ops::RealDiv, {1.5, 3, 1, 2, 1.5, 3, 1, 2});
+  ADD_OP("FloorDiv", ops::FloorDiv, {1, 3, 1, 2, 1, 3, 1, 2});
+  ADD_OP("Minimum", ops::Minimum, {2, 2, 3, 3, 2, 2, 3, 3});
+  ADD_OP("Maximum", ops::Maximum, {3, 6, 3, 6, 3, 6, 3, 6});
+  ADD_OP("Pow", ops::Pow, {9, 36, 27, 216, 9, 36, 27, 216});
+#undef ADD_OP
+  std::vector<string> ops_to_test;
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:2512]: We don't need this, we can iterate through an unordered map instead.

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc (hunk 3 above)
Comment (tfeher) [reply tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:2514]: done

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc (hunk 2 above)
Comment (tfeher) [reply tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:2496]: Renamed to op_test_info

**Review by tfeher — COMMENTED** [review]
Thanks @Bixia1 for the review, I have addressed the issues.

**Review by bixia1 — APPROVED** [review]

**Pull request merged**